### DISCUSSION TITLE
GP2040 mame and retroarch use

### DISCUSSION THREAD

**Discussion by R0lleston** [OpenStickCommunity/GP2040-CE#280] [General]
Hi all,

is the GP2040-CE an alternative to the IPAC arcade button encoder by Ultimarc or just a device primarily for fighting games.

I’m building an arcade cabinet which uses Mame and Retroarch and interfaces with the standard joystick and push buttons, but want a configurable set up that the IPAC set up allows.

Also does this project offer a shift mode, where holding a key and pressing another gives extra commands ?

Sorry if this is a trivial question, but of a newbie here.

**Comment by bsstephan** [comment]
I don't know what the IPAC is, but I think GP2040-CE would work just fine for a MAME + RetroArch setup. It can act as a couple standard input devices including a keyboard, depending on your goals.

GP2040-CE v0.7.1 supports a handful of hotkeys that, in a couple cases, sends a different button to the host. I am working on a PR that will make more hotkeys configurable by the user, and perhaps with time that can be expanded to allow the sending of more keys/buttons down.

**Reply by R0lleston** [reply]
@bsstephan thanks for the reply
As an fyi here is the link for the item I was referring to
https://www.ultimarc.com/control-interfaces/i-pacs/i-pac2/

**Comment by bsstephan** [comment]
On June 8, 2023 7:50:37 PM CDT, R0lleston ***@***.***> wrote:
>@bsstephan thanks for the reply
>As an fyi here is the link for the item I was referring to
>https://www.ultimarc.com/control-interfaces/i-pacs/i-pac2/
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/orgs/OpenStickCommunity/discussions/280#discussioncomment-6127757
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
Ah, got it. So I'd say that we could be used for such a setup, but we don't offer that many inputs, and I'm not sure how a trackball/spinner would interact with our firmware's analogs. I would also point out that our hotkey support isn't intended to be a full-on shift with a new array of outputs, but maybe one day that could be achievable.

So doable but with some compromises, I'd say. Code contributions welcome. :)
